packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/eventbridge-scheduler/create-schedule.ts

```diff
@@ -0,0 +1,260 @@
+}
+
+/**
+ * Create a new AWS EventBridge Scheduler schedule
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/APIReference/API_CreateSchedule.html
+ */
+export class EventBridgeSchedulerCreateScheduleTask extends sfn.TaskStateBase {
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    super(scope, id, props);
+
+    this.validateProps(props);
+
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: [
+          Stack.of(this).formatArn({
+            service: 'scheduler',
+            resource: 'schedule',
+            resourceName: `${this.props.groupName ?? 'default'}/${this.props.scheduleName}`,
+          }),
+        ],
+        actions: [
+          'scheduler:CreateSchedule',
+        ],
+      }),
+      new iam.PolicyStatement({
+        resources: [props.role.roleArn],
+        actions: [
+          'iam:PassRole',
+        ],
+      }),
+    ];
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('aws-sdk:scheduler', 'createSchedule', this.integrationPattern),
+      Parameters: {
+        ActionAfterCompletion: this.props.actionAfterCompletion,
+        ClientToken: this.props.clientToken,
+        Description: this.props.description,
+        EndDate: this.props.endDate ? this.props.endDate.toISOString() : undefined,
+        FlexibleTimeWindow: {
+          Mode: this.props.flexibleTimeWindow ? 'FLEXIBLE' : 'OFF',
+          MaximumWindowInMinutes: this.props.flexibleTimeWindow?.toMinutes(),
+        },
+        GroupName: this.props.groupName,
+        KmsKeyArn: this.props.kmsKey?.keyArn,
+        Name: this.props.scheduleName,
+        ScheduleExpression: this.props.scheduleExpression,
+        ScheduleExpressionTimezone: this.props.timezone,
+        StartDate: this.props.startDate ? this.props.startDate.toISOString() : undefined,
+        State: (this.props.enabled ?? true) ? 'ENABLED' : 'DISABLED',
+        Target: {
+          Arn: this.props.target,
+          RoleArn: this.props.role.roleArn,
+          Input: this.props.input,
+          RetryPolicy: this.props.retryPolicy ? {
+            MaximumEventAgeInSeconds: this.props.retryPolicy.maximumEventAge.toSeconds(),
+            MaximumRetryAttempts: this.props.retryPolicy.maximumRetryAttempts,
+          } : undefined,
+          DeadLetterConfig: this.props.deadLetterQueue ? {
+            Arn: this.props.deadLetterQueue.queueArn,
+          } : undefined,
+        },
+      },
+    };
+  }
+
+  private validateProps(props: EventBridgeSchedulerCreateScheduleTaskProps) {
+    if (
```

Comment:
We should also apply validation according to the [specified pattern](https://docs.aws.amazon.com/scheduler/latest/APIReference/API_CreateSchedule.html).

Reply:
I've added some validations.
- clientToken format ([a-zA-Z0-9-_]+$)
- groupName format ([a-zA-Z0-9-_.]+$)
- timezone length (1-50)